crates/types/src/task_manager.rs

```diff
@@ -123,6 +123,20 @@ impl TaskSpawner {
         }
     }
 
+    /// Spawns a non-critical, blocking task on tokio and records the JoinHandle and name.
+    ///
+    /// Other tasks are unaffected when this task resolves.
+    pub fn spawn_blocking_task<F, S: ToString>(&self, name: S, task: F)
```

Comment:
Nice.  I don't think these tasks will respond to aborts, etc but also we have timeouts so I don't think that matters.